internal/sink/sink.go

```diff
@@ -640,7 +654,16 @@ func (s *Sink) ParseMessage(e *kafka.Message) (string, *serializer.CloudEventsKa
 }
 
 func (s *Sink) Close() error {
-	s.config.Logger.Info("closing sink")
+	logger := s.config.Logger.With("operation", "close")
+
+	logger.Info("closing sink")
+
+	logger.Debug("acquiring lock to prevent closing sink during flush operation")
```

Comment:
Please add comment while this is necessary

Reply:
Updated the PR with comments where locking is used. Ty